sycl/include/CL/sycl/detail/device_impl.hpp

```diff
@@ -74,17 +74,19 @@ class device_impl {
   /// Check if device is a CPU device
   ///
   /// @return true if SYCL device is a CPU device
-  bool is_cpu() const { return (MType == PI_DEVICE_TYPE_CPU); }
+  bool is_cpu() const { return (!is_host() && (MType == PI_DEVICE_TYPE_CPU)); }
```

Comment:
Not sure it's right approach. Probably the right approach would be to add new device type(host) to _pi_device_type enum. What do you think?

Reply:
I don't think PI wants to cover "host". It is rather interface for offloading. So I think what you are doing is preferable, but is it really OK that we return false in all of these for "host"? Should we return true for is_cpu()?

Reply:
@smaslov-intel technically, host device is not CPU, it is host. But I agree with your point about covering host on PI.

Reply:
OK, then I am OK with the fix.